docs/diagrams/data-flow/data-flow-api-endpoint.puml

```diff
@@ -0,0 +1,153 @@
+'
+'  Copyright (c) 2021 Daimler TSS GmbH
+'
+'  This program and the accompanying materials are made available under the
+'  terms of the Apache License, Version 2.0 which is available at
+'  https://www.apache.org/licenses/LICENSE-2.0
+'
+'  SPDX-License-Identifier: Apache-2.0
+'
+'  Contributors:
+'      Daimler TSS GmbH - Initial Version
+'
+
+@startuml
+
+skinParam NoteBackgroundColor WhiteSmoke
+skinParam NoteFontColor Black
+skinParam ParticipantBackgroundColor WhiteSmoke
+skinParam ActorBackgroundColor WhiteSmoke
+skinParam AgentBackgroundColor White
+skinParam AgentBorderColor SkyBlue
+skinparam shadowing false
+
+!define ConsumerColor f8f2ff
+!define ProviderColor d9edff
+!define WarningColor Business
+!define LeadColor Technology
+
+autonumber
+
+box Consumer #ConsumerColor
+    participant CBE as "Backend Service"
+    participant CCP as "Control Plane" #Business
+    participant CDP as "Data Plane" #Business
+end box
+
+box Provider #ProviderColor
+    participant PDP as "Data Plane" #Business
+    participant PCP as "Control Plane" #Business
+    participant PBE as "Backend Service"
+end box
+
+title
+ **Access to REST API**
+ <sub>Control Plane and Date Plane are part of the EDC.</sub>
+ <sub>Precondition: contract 'access to api' iuncluding constraints (e.g. how often, for which purpose etc.) is signed</sub>
+
+ Case 1a: Each API call matches one TransferProcess (http push)
+ <sub>Provider is pushing data to the Consumer</sub>
+end title
+
+CBE ->> CCP : getAsset(query params, int destination)
+activate CCP
+CCP ->> CDP : provision ext endpoint
+CCP ->> PCP : start data transfer(asset, contract, query params, ext endpoint)
+activate PCP
+PCP ->> PBE : provision data\n(if needed)
+PCP ->> PDP : start data transfer
+activate PDP
+PDP -> PBE : pull data
+activate PBE
+return data
+PDP -> CDP : push data to ext endpoint
+activate CDP
+CDP -> CBE : push data to int destination
+note left of CDP : Dataplane\nacts as proxy.
+PDP ->> PCP : notify: data pushed
+deactivate PDP
+CDP ->> CCP : notify: data received
+deactivate CDP
+CCP ->> CDP : deprovision
+PCP ->> PBE : deprovision\n(if needed)
+deactivate CCP
+deactivate PCP
+
+
+
+newpage Case 1b: Each API call matches one TransferProcess (http pull)\n<sub>Consumer is pulling data from Provider</sub>
```

Comment:
In step 6 the Backend Data Service would actually call the end of the data plane to proxy the data through to the receiving data plane. Same in case of streaming - the data plane of the consumer has to wait as the DP will no know when date is finally provided.

Reply:
hm, interesting question.  Would make sense if more than one call will be made to the provider, similar to diagram 2.
Wonder if it is sufficient to have it like shown in diagram 2 or if we a mix, i.e. a data plane that is actively copying data instead of just being a proxy.